frontend/src/components/Profile/Profile.js

```diff
@@ -1,10 +1,85 @@
 import UserInfo from "./UserInfo";
+import {
+  Container,
+  Card,
+  Box,
+  CardContent,
+  List,
+  ListItem,
+  Typography,
+  Grid,
+} from "@material-ui/core";
+import UserGroups from "../Groups/UserGroups";
+import { connect } from "react-redux";
+import { useEffect } from "react";
+import { styled } from "@material-ui/styles";
+import Event from "../Events/Event";
+import { getEvents } from "../../actions/events";
+import UserEvents from "../Events/UserEvents";
+import { getHomePageData } from "../../actions/user";
+import TmpComponent from "./TmpComponent";
+import { useAuth } from "../../context/AuthContext";
+
+const fromTop = "60px";
+
+const SCBox = styled(Box)({
+  display: "flex",
+});
+
+const Box2 = styled(Box)({
+  display: "flex",
+  margin: fromTop,
+});
+
+const Box3 = styled(Box)({
+  display: "flex",
+  marginTop: fromTop,
+  marginLeft: "0px",
+});
+
+function Profile(props) {
+  const { getHomePageData } = props;
+  const { currentUser } = useAuth();
+
+  useEffect(() => {
+    getHomePageData(currentUser.uid);
+  }, [getHomePageData, currentUser.uid]);
 
-function Profile() {
   return (
-    <div className="profile-container">
+    <SCBox>
       <UserInfo />
-    </div>
+      <Grid>
+        <Box3>
+          <TmpComponent
+            title="Your Joined Events"
+            events={props.joinedEvents}
+          />
+          <TmpComponent
+            title="Your Created Events"
+            events={props.createdEvents}
+          />
+        </Box3>
+      </Grid>
+      <Grid>
+        <Box2>
+          <UserGroups />
+        </Box2>
+      </Grid>
+    </SCBox>
   );
 }
-export default Profile;
+
+const mapStateToProps = (state) => {
+  return {
+    createdEvents: state.user.userEvents.created,
+    joinedEvents: state.user.userEvents.joined,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    getHomePageData: (id) => dispatch(getHomePageData(id)),
+  };
```

Comment:
I believe you're suppose to make a new action like getProfilePageData(id of the profile), that performs getUserProfile and getUserProfileEvents, and when it reaches reducer it should update the profile state, not the user state (which is the current user state). The profile can be the profile of a different user, so you shouldn't call getHomePageData(), which updates current user state and you shouldn't call with, line 45 `getHomePageData(currentUser.uid);` and instead with useParam().